tests/benchmarks/sys_kernel/testcase.yaml

```diff
@@ -1,5 +1,5 @@
 tests:
   benchmark.kernel.core:
     arch_exclude: nios2 riscv32 xtensa
-    min_ram: 32
+    min_ram: 36
```

Comment:
main issue here is that you will be skipping this for more boards than just twr_ke18f, so my suggestion is to first try to make this fit on twr_ke18f, maybe we are enabling too many things on that board which increases the ram usage, second and if the above fails, just exclude it using platfrom_exclude.

This way we do not disable all boards that have less than 36 which can be alot, given that many boards have 32.

Reply:
I have already gone through the enabled options for the twr_ke18f; nothing can be reduced in this regard. What can be reduced (and this will benefit all boards), is the amount of RAM used for the `mem_slap.c` tests as suggested above.

Reply:
right, based on that, I propose this:

```diff
diff --git a/tests/benchmarks/sys_kernel/CMakeLists.txt b/tests/benchmarks/sys_kernel/CMakeLists.txt
index 093204a5ba..c32d12436a 100644
--- a/tests/benchmarks/sys_kernel/CMakeLists.txt
+++ b/tests/benchmarks/sys_kernel/CMakeLists.txt
@@ -4,5 +4,9 @@ cmake_minimum_required(VERSION 3.13.1)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(sys_kernel)

+if(DEFINED NUMBER_OF_LOOPS)
+  zephyr_compile_definitions(NUMBER_OF_LOOPS=${NUMBER_OF_LOOPS})
+endif()
+
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
diff --git a/tests/benchmarks/sys_kernel/src/syskernel.h b/tests/benchmarks/sys_kernel/src/syskernel.h
index 4284de6a0d..fdd838f0df 100644
--- a/tests/benchmarks/sys_kernel/src/syskernel.h
+++ b/tests/benchmarks/sys_kernel/src/syskernel.h
@@ -15,7 +15,9 @@
 #include <toolchain.h>

 #define STACK_SIZE 2048
+#ifndef NUMBER_OF_LOOPS
 #define NUMBER_OF_LOOPS 1000
+#endif

 extern K_THREAD_STACK_DEFINE(thread_stack1, STACK_SIZE);
 extern K_THREAD_STACK_DEFINE(thread_stack2, STACK_SIZE);
diff --git a/tests/benchmarks/sys_kernel/testcase.yaml b/tests/benchmarks/sys_kernel/testcase.yaml
index ab1af06676..6ee951f5e1 100644
--- a/tests/benchmarks/sys_kernel/testcase.yaml
+++ b/tests/benchmarks/sys_kernel/testcase.yaml
@@ -1,5 +1,11 @@
+common:
+  min_ram: 32
+  tags: benchmark
 tests:
   benchmark.kernel.core:
     arch_exclude: nios2 riscv32 xtensa
-    min_ram: 32
-    tags: benchmark
+    platform_exclude: twr_ke18f
+  benchmark.kernel.core.lowmem:
+    platform_allow: twr_ke18f
+    extra_args: "NUMBER_OF_LOOPS=100"
+
```

Reply:
or simpler if you want...

```
diff --git a/tests/benchmarks/sys_kernel/src/syskernel.h b/tests/benchmarks/sys_kernel/src/syskernel.h
index fdd838f0df..c0082a8c41 100644
--- a/tests/benchmarks/sys_kernel/src/syskernel.h
+++ b/tests/benchmarks/sys_kernel/src/syskernel.h
@@ -15,7 +15,9 @@
 #include <toolchain.h>

 #define STACK_SIZE 2048
-#ifndef NUMBER_OF_LOOPS
+#if CONFIG_SRAM_SIZE <= 32
+#define NUMBER_OF_LOOPS 100
+#else
 #define NUMBER_OF_LOOPS 1000
 #endif
```